### DISCUSSION TITLE
Using Plurals in t Macro

### DISCUSSION THREAD

**Discussion by crazyyi** [lingui/js-lingui#1508] [Q&A] [answered]
I have a requirement to display days based on the numbers and others if the variable doesn't contain the text "day". The div below is for each plan in a given array.

I have tried this:

`num = plan.name.trim().split(" ")[0]`
```
<div>{plan.name.includes("day")? t({id: `planname${index}`, message: plural(num, {one: "# day", other: "# days"})}) : plan.name}<div>
```

It's getting complex because it has both a ternary operator and plural form, and because the **div** is inside a map function, I have to put **${index}** in the id so that it displays the correct plural form otherwise every plan would just display "days". 

What I was getting in the json file is:

`"planname": "{0, plural, one {day} other {days}}"`

But if I change `day` and `days` to some other characters in another locale the translation wouldn't change. Not sure if it's caused by the ternary operator, because it would always show days for every plan if I switch the locale.

After a few more tweaking, I got interesting results. I removed the  **${index}** in the id, and the translations in other locale work fine. Then the plural form in the default language (en) doesn't work anymore, it would always display "days" for each plan. But it works for another language (e.g. fr) which doesn't even have a given translation.  Is there any solution to make both of them work?

**Comment by timofei-iatsenko** [comment, marked as answer]
> t({id: `planname${index} `, message: plural(num, {one: "day", other: "days"})})

That is not going to work. All values, such as id, should be statically analyzable. `planname${index} ` is not statically analyzable.

> But it works for another language (e.g. fr) which doesn't even have a given translation. Is there any solution to make both of them work?

Look for the issue in your app code. It's misusage somewhere.

**Reply by crazyyi** [reply]
Thanks I just found the problem. I didn't trim the num variable. The `num` is a string, not a number.
But looks like `{one: "# day", other: "# days"} `sometimes get NaN value. I need to concat the num variable with the plural form to make it work. Like this:
```
num.trim() + " " + t({id: `planname` , message: plural(num.trim(), {one: "day", other: "days"})})
```

**Reply by timofei-iatsenko** [reply]
again, look for the issue in your code, `{one: "# day", other: "# days"}` this should work well. If it returns NaN it means your number is NaN...
